### DISCUSSION TITLE
Changing the Default Boot option from Internal HDD to a different ipxe boot location

### DISCUSSION THREAD

**Discussion by Theknight2015** [antonym/netboot.xyz#1326] [Q&A]
So I'm new to using netboot.xyz and I wanted to ask what may be a dumb question. I have several PC's with internal HDD's and I also have several PC's that use a current ipxe boot sever to boot a custom OS specifically meant for remote desktop. I would like to start using netboot.xyz on my network as the default ipxe boot server so that I may use the tools on the PC's with internal HDD's. My problem arises when I try to boot the other PC's that are supposed to be used for the remote systems. They boot to netboot.xyz and then they attempt to boot to their internal HDD's as the default option lists but those systems do not have internal HDD's. 

I would like to know if it is possible to set the default boot option to another ipxe server address and file name so that once the default timeout occurs, it boots to the second ipxe server but if you change the option from "default" to any other netboot.xyz boot option before that default timeout occurs then the default timeout countdown stops and you can take as much time as you want to select the appropriate netboot.xyz boot option/tool that you're looking for.

I also changed the default timeout option in boot.cfg from 300,000 to 60,000 because I'm assuming it's in milliseconds and 300,000 milliseconds is 5 minutes where as 60,000 milliseconds is 60 seconds. If I'm wrong on this, please correct me.

Any help on figuring out a way to change the default boot option from internal HDD to a different ipxe server address and file name would be greatly appreciated.

**Comment by antonym** [comment]
There are a few options you might be able to try. If you are booting netboot.xyz from your local network, you can add some configuration files that can override the menu and kick off a process.

https://github.com/netbootxyz/netboot.xyz/blob/development/roles/netbootxyz/templates/disks/netboot.xyz.j2#L98

```
chain tftp://${tftp-server}/local-vars.ipxe || echo ${VARS_ERR}
isset ${hostname} && chain --autofree tftp://${tftp-server}/HOSTNAME-${hostname}.ipxe || echo Custom boot by Hostname not found trying MAC...
chain --autofree tftp://${tftp-server}/MAC-${mac:hexraw}.ipxe || echo Custom boot by MAC not found booting default...
chain --autofree tftp://${tftp-server}/menu.ipxe || echo ${TFTP_ERR} && goto menu
```

These are all hooks you can put in place to change the behavior of the menu. If you are booting from your local tftp server, you should be able to see these pop up. With this, you should be able to change the behavior by mac address, or set the hostname that is set by DHCP.

**Reply by Theknight2015** [reply]
Okay so that definitely seems a little over my head. I've looked at iPXE boot software in the past and read up a little on chain boot scripts but I'm not entirely familiar with how they operate or what they do. I looked at the link you posted as well. That seems to be a boot file that tells the system what menu to display? I suppose my question now is - How to I make a custom menu or can I simply change the default boot option on the default menu to point elsewhere and if so, how do I change that?

Or if I'm missing the point of what you said, please tell me.
